apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.cs

```diff
@@ -53,5 +53,130 @@ public void EqualsTest(SpannerDbType spannerType1, SpannerDbType spannerType2)
+            yield return new object[] { "  BYTES  ", SpannerDbType.Bytes };
+            yield return new object[] { "  DATE  ", SpannerDbType.Date };
+            yield return new object[] { "  FLOAT64  ", SpannerDbType.Float64 };
+            yield return new object[] { "  INT64  ", SpannerDbType.Int64 };
+            yield return new object[] { "  TIMESTAMP  ", SpannerDbType.Timestamp };
+
+            yield return new object[] { "STRING(2)", SpannerDbType.String.WithSize(2) };
+            yield return new object[] { "STRING(100)", SpannerDbType.String.WithSize(100) };
+            yield return new object[] { " STRING ( 100  )  ", SpannerDbType.String.WithSize(100) };
+            yield return new object[] { "STRING (  MAX )", SpannerDbType.String };
+            yield return new object[] { "STRING(100)", SpannerDbType.String, false };
+            yield return new object[] { "BYTES(100)", SpannerDbType.Bytes.WithSize(100) };
+            yield return new object[] { "BYTES(100)", SpannerDbType.Bytes, false };
+            yield return new object[] { " BYTES ( 100  )  ", SpannerDbType.Bytes.WithSize(100) };
+
+            //some common array types.
+            yield return new object[] { "ARRAY<STRING>", SpannerDbType.ArrayOf(SpannerDbType.String) };
+            yield return new object[] { "ARRAY<STRING(MAX)>", SpannerDbType.ArrayOf(SpannerDbType.String) };
+            yield return new object[] { "ARRAY<STRING(5)>", SpannerDbType.ArrayOf(SpannerDbType.String.WithSize(5)) };
+            yield return new object[] { "ARRAY<STRING>", SpannerDbType.ArrayOf(SpannerDbType.String) };
+            yield return new object[] { "ARRAY<BOOL>", SpannerDbType.ArrayOf(SpannerDbType.Bool) };
+            yield return new object[] { "ARRAY<BYTES>", SpannerDbType.ArrayOf(SpannerDbType.Bytes) };
+            yield return new object[] { "ARRAY<BYTES(100)>", SpannerDbType.ArrayOf(SpannerDbType.Bytes.WithSize(100)) };
+            yield return new object[] { "ARRAY<DATE>", SpannerDbType.ArrayOf(SpannerDbType.Date) };
+            yield return new object[] { "ARRAY<FLOAT64>", SpannerDbType.ArrayOf(SpannerDbType.Float64) };
+            yield return new object[] { "ARRAY<INT64>", SpannerDbType.ArrayOf(SpannerDbType.Int64) };
+            yield return new object[] { "ARRAY<TIMESTAMP>", SpannerDbType.ArrayOf(SpannerDbType.Timestamp) };
+
+            yield return new object[] { "ARRAY<STRING(5)>", SpannerDbType.ArrayOf(SpannerDbType.String), false };
+            yield return new object[] { "ARRAY<BYTES(5)>", SpannerDbType.ArrayOf(SpannerDbType.Bytes), false };
+
+            yield return new object[] { "ARRAY <  STRING (   5 )>", SpannerDbType.ArrayOf(SpannerDbType.String.WithSize(5)) };
+            yield return new object[] { "ARRAY<  STRING (   5 )  > ", SpannerDbType.ArrayOf(SpannerDbType.String.WithSize(5)) };
+
+            //simple structs
+            yield return new object[] { "STRUCT<F1:STRING,F2:INT64>", SpannerDbType.StructOf(
```

Comment:
there are cases where from the spanner side the field values of a struct are given in the order of the struct field declarations (eg Type=Struct {field1=string, field2=int...} Values=list<Value> {field1Value...}} -- so internally we need to preserve the ordinals when converting Value to the clr type version.  I do not believe there is any way we'd do this conversion on a user specified SpannerDbType that doesn't have this info (its always constructed from the V1.Type from the server).
The reason I didn't include ordering in equality is that I felt it would be a big burden and somewhat confusing as its exposing whats really an implementation detail of struct communication than anything else.
I'll double check whether there is a potential of error here.